friktion/friktion/config.py

```diff
@@ -0,0 +1,168 @@
+from decimal import Decimal
+from typing import Any
+
+from asgiref.sync import async_to_sync
+from solana.publickey import PublicKey
+
+from friktion.bid_details import BidDetails
+from friktion.offer import Offer
+from friktion.swap import Network, SwapContract
+from sdk_commons.chains import Chains
+from sdk_commons.config import BidValidation, OfferDetails, OfferTokenDetails, SDKConfig
+
+# from friktion.friktion_anchor.accounts.swap_order import SwapOrder
+
+
+class AuthorizationPages:
+    mainnet = "https://app.friktion.fi/approve"
+    testnet = "https://devnet.friktion.fi/approve"
+
+
+# TODO: remove unused parameters (for example rpc_uri and nonce)
+
+
+class FriktionSDKConfig(SDKConfig):
+
+    authorization_pages = AuthorizationPages
+    supported_chains = [Chains.SOLANA_DEV]
+
+    CHAIN_NETWORK_MAP = {
+        Chains.SOLANA_DEV: Network.DEVNET,
+        Chains.SOLANA_MAIN: Network.MAINNET,
+    }
+
+    def create_offer(
+        self,
+        contract_address: str,
+        chain_id: int,
+        rpc_uri: str,
+        oToken: str,
+        bidding_token: str,
+        min_price: int,
+        min_bid_size: int,
+        offer_amount: int,
+        public_key: str,
+        private_key: str,
+        *args: Any,
+        **kwargs: Any,
+    ) -> str:
+        """Create an offer"""
+
+        # swap_order = SwapOrder(...)
+        # Offer.from_swap_order(
+        #     swap_order=swap_order,
+        #     address=PublicKey(public_key)
+        # )
+
+        return "not-implemented-yet"
+
+    def get_otoken_details(
+        self,
+        contract_address: str,
+        chain_id: int,
+        rpc_uri: str,
+        offer_id: int,
+        seller: str,
+        *args: Any,
+        **kwargs: Any,
+    ) -> OfferTokenDetails:
+        """Return details about the offer token"""
+
+        network = self.CHAIN_NETWORK_MAP[Chains(chain_id)]
+        swap_contract = SwapContract(network)
+        details: OfferTokenDetails = async_to_sync(swap_contract.get_offered_token_details)(
+            PublicKey(seller), offer_id
+        )
+
+        return details
+
+    def get_offer_details(
+        self,
+        contract_address: str,
+        chain_id: int,
+        rpc_uri: str,
+        offer_id: int,
+        seller: str,
+        *args: Any,
+        **kwargs: Any,
+    ) -> OfferDetails:
+        """Return details for a given offer"""
+
+        network = self.CHAIN_NETWORK_MAP[Chains(chain_id)]
+        swap_contract = SwapContract(network)
+
+        details: Offer = async_to_sync(swap_contract.get_offer_details)(
+            PublicKey(seller), offer_id
+        )
+        return {
+            'seller': str(details.seller),
+            'biddingToken': str(details.biddingToken),
+            'minPrice': str(details.minPrice),
```

Comment:
Not really. We need to cast it to str only because minPrice is currently [annotated to be a string](https://github.com/tradeparadigm/sdks/blob/6e04683eeb17b24fa7ff71f5baaf2ae8f7a6b001/sdk_commons/sdk_commons/config.py#L13). We can easily change it, but which type should be? (also considering the other venues).
int? float? Decimal?

Reply:
Oh got it. It is not a Decimal at this point as we still have to apply the resolution. So I think we can leave that to adjust as the broader discussion about value resolution goes